### PR TITLE
Improvement to regexps

### PR DESCRIPTION
**Pull request by remram44**
This improves the regexps somewhat, and avoids their use for URLs by using the built-in [URL class](https://developer.mozilla.org/en-US/docs/Web/API/URL).

### CODE REVIEW

File: extension/updateswh.js [hunk 1]
```diff
@@ -211,34 +210,34 @@ function setupGitLabInstance(url, pattern, type) {
...
+        reject:  /^https?:\/\/(0xacab\.org|gite\.lirmm\.fr|framagit\.org|gricad-gitlab\.univ-grenoble-alpes\.fr)\/users\/sign_in/,
         type: 'GitLab instance',
         handler: setupGitLabInstance
     },
     // heuristic: we handle gitlab.*.* as a GitLab instance
     {
-        pattern: /^https?:\/\/gitlab.[^.]*.[^.\/]*\/[^\/]*\/[^\/]+/,
-        reject:  /^https?:\/\/gitlab.[^.]*.[^.\/]*\/users\/sign_in/,
+        pattern: /^https?:\/\/gitlab\.[^.\/]+\.[^.\/]+\/[^\/]+\/[^\/]+/,
+        reject:  /^https?:\/\/gitlab\.[^.\/]+\.[^.\/]+\/users\/sign_in/,
```
Comment (remram44) [inline extension/updateswh.js:240]: For example the regexp would incorrectly accept `'https://gitlab.com/remram44/remram.fr/tree/master'` (the first `[^.]*` would match `com/remram44/remram`)

Comment (rdicosmo) [reply extension/updateswh.js:240]: Thanks a lot @remram44  for spotting and fixing all this!

File: extension/updateswh.js [hunk 2]
```diff
@@ -142,7 +142,7 @@ function testupdateforge(url, forgespecs) {
 
 function setupGitHub(url, pattern, type) {
     var projecturl = pattern.exec(url)[0]; // this is the url of the project
-    var userproject = projecturl.replace(/^https?:\/\/github\.com\//, ""); // this is the user+project fragment
+    var userproject = new URL(projecturl).pathname.substring(1); // this is the user+project fragment
```
Comment (rdicosmo) [inline extension/updateswh.js:145]: Thanks for the URL class! Why we need the substring(1) here?

Comment (remram44) [reply extension/updateswh.js:145]: `pathname` will include the first slash, so it's `/user/project`. `pathname.substring(1)` is `user/project`

Comment (rdicosmo) [reply extension/updateswh.js:145]: Ok, got it.

**Pull request merged**